src/search/tasks/default_value_axioms_task.cc

```diff
@@ -388,13 +389,16 @@ int DefaultValueAxiomsTask::get_num_axioms() const {
     return parent->get_num_axioms() + default_value_axioms.size();
 }
 
+AxiomEvaluator &DefaultValueAxiomsTask::get_axiom_evaluator() const {
+    return *axiom_evaluator;
+}
+
 shared_ptr<AbstractTask> get_default_value_axioms_task_if_needed(
     const shared_ptr<AbstractTask> &task,
     AxiomHandlingType axioms) {
     TaskProxy proxy(*task);
     if (task_properties::has_axioms(proxy)) {
-        return make_shared<tasks::DefaultValueAxiomsTask>(
-            DefaultValueAxiomsTask(task, axioms));
+        return make_shared<tasks::DefaultValueAxiomsTask>(task, axioms);
```

Comment:
I don't understand this class well enough to review this part of the pull request.

Reply:
This particular change looks harmless to me: it's just getting rid of an unnecessary copy.

Other than that, the current change is to not delegate the axiom evaluator here and instead create one from scratch. From what I understood from the discussion with Gabi about this, this is a mistake: Tasks of this kind have no well-defined semantics for the axiom evaluation and the new axioms are only meant to be used by the heuristic not to generate states. In the old code, this is only enforced with documentation, with the `get_axiom_evaluator` method, we could throw an error if someone tries to access an axiom evaluator for a task like this.